### PR TITLE
include commit in pre-release builds

### PR DESCRIPTION
**Pull request by BenTheElder**
xref: #986

### CODE REVIEW

**Comment by BenTheElder** [comment]
later we'll also use this in kubernetes CI to record the kind commit used to test.

**Comment by BenTheElder** [comment]
/cc @amwat 
```
[bentheelder@bentheelder:~/go/src/sigs.k8s.io/kind·2019-10-21T15:37:22-0700·include-commit@dfbd929e]
$ make install
hack/go_container.sh go build -v -o /out/kind -ldflags "-X sigs.k8s.io/kind/cmd/kind/version.GitCommit=dfbd929e5ff1e665b07ff9bd46d966d281c51b4c"
install -d /usr/local/google/home/bentheelder/go/bin
install /usr/local/google/home/bentheelder/go/src/sigs.k8s.io/kind/bin/kind /usr/local/google/home/bentheelder/go/bin/kind

[bentheelder@bentheelder:~/go/src/sigs.k8s.io/kind·2019-10-21T15:45:20-0700·include-commit@dfbd929e]
$ kind version
v0.6.0-alpha+dfbd929e5ff1e665b07ff9bd46d966d281c51b4c
```

File: cmd/kind/version/version.go [hunk 1]
```diff
@@ -23,19 +23,41 @@ import (
...
-const Version = "v0.6.0-alpha"
+// Version returns the kind CLI Semantic Version
+func Version() string {
+	v := VersionCore
+	// add pre-release version info if we have it
+	if VersionPreRelease != "" {
+		v += "-" + VersionPreRelease
+		// if commit was set, add the + <build>
+		// we only do this for pre-release versions
+		if GitCommit != "" {
```
Comment (BenTheElder) [inline cmd/kind/version/version.go]: to consider in a follow-up: we might want to instead (or additionally) have the release scripts to `export COMMIT=""` to make it easier to reproducibly build the binaries from a source tarball. or instructions for doing `COMMIT=<release-commit> make build`

Comment (BenTheElder) [reply cmd/kind/version/version.go]: I think it is probably better to include the commit, and then if someone wants to reproduce the binaries without git they can set `COMMIT=` (the makefile should already support this)

File: Makefile [hunk 2]
```diff
@@ -28,6 +28,9 @@ INSTALL?=install
 # make install will place binaries here
 # the default path attempts to mimic go install
 INSTALL_DIR?=$(shell hack/build/goinstalldir.sh)
+# record the source commit in the binary
+COMMIT?=$(shell git rev-parse HEAD 2>/dev/null)
```
Comment (aojea) [inline Makefile]: Why not the short commit i

Comment (aojea) [reply Makefile]: It's easier to read imho

Comment (BenTheElder) [reply Makefile]: The short version is just this truncated. We can truncate at runtime when formatting if we want, but this is the correct unambiguous commit ID.

Comment (BenTheElder) [reply Makefile]: https://github.com/kubernetes-sigs/kind/pull/995

**Pull request merged**